routers/api/v1/admin/user.go

```diff
@@ -236,3 +236,55 @@ func CreatePublicKey(ctx *context.APIContext, form api.CreateKeyOption) {
 	}
 	user.CreateUserPublicKey(ctx, form, u.ID)
 }
+
+// DeleteUserPublicKey api for deleting a user's public key
+func DeleteUserPublicKey(ctx *context.APIContext) {
+	// swagger:operation DELETE /admin/users/{username}/keys/{id} admin adminDeleteUserPublicKey
+	// ---
+	// summary: Delete a user's public key on behalf of a user
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: username
+	//   in: path
+	//   description: username of user
+	//   type: string
+	//   required: true
+	// - name: id
+	//   in: path
+	//   description: key's id to delete
+	//   type: string
+	//   required: true
+	// responses:
+	//   "204":
+	//     "$ref": "#/responses/empty"
+	//   "404":
+	//     "$ref": "#/responses/notFound"
+	//   "403":
+	//     "$ref": "#/responses/forbidden"
+	u := user.GetUserByParams(ctx)
+	if ctx.Written() {
+		return
+	}
+
+	if _, err := models.GetPublicKeyByID(ctx.ParamsInt64(":id")); err != nil {
```

Comment:
It would be nice if we didn't have to call `GetPublicKeyByID` before calling `DeletePublicKey`, since `DeletePublicKey` already calls `GetPublicKeyByID`. Perhaps it's worthwhile to have `DeletePublicKey` return a `ErrKeyNotExist` when you try to delete a non-existent key?

Reply:
That function is used only in one file, so additional checks for that could be added for that error